### PR TITLE
VACMS-13278: patch to address menu link being too short

### LINKED ISSUE #13278: Menu links incorrectly limited to 255 characters
## Describe the defect
Slack reference https://dsva.slack.com/archives/CDHBKAL9W/p1681401208207779

Edited 5/23/23 to make this complete
When trying to update several urls in the megamenu. One of them keeps returning an error in Drupal.

For VA Leadership it should be this url: https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff

## To Reproduce
1. Go to '/admin/structure/menu/item/934/edit?destination=/admin/structure/menu/manage/header-megamenu'
2. paste the following url in the link field 'https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff'
3. Click save
image
5. Go to ` /admin/reports/dblog` to see the  complete errors
image
```
Drupal\Core\Entity\EntityStorageException: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'url' at row 1: UPDATE "menu_tree" SET "menu_name"=:db_update_placeholder_0, "url"=:db_update_placeholder_1, "route_name"=:db_update_placeholder_2, "route_parameters"=:db_update_placeholder_3, "options"=:db_update_placeholder_4, "title"=:db_update_placeholder_5, "description"=:db_update_placeholder_6, "weight"=:db_update_placeholder_7, "enabled"=:db_update_placeholder_8, "expanded"=:db_update_placeholder_9, "parent"=:db_update_placeholder_10, "id"=:db_update_placeholder_11, "class"=:db_update_placeholder_12, "provider"=:db_update_placeholder_13, "metadata"=:db_update_placeholder_14, "form_class"=:db_update_placeholder_15, "has_children"=:db_update_placeholder_16, "discovered"=:db_update_placeholder_17, "depth"=:db_update_placeholder_18, "p1"=:db_update_placeholder_19, "p2"=:db_update_placeholder_20, "p3"=:db_update_placeholder_21, "p4"=:db_update_placeholder_22, "p5"=:db_update_placeholder_23, "p6"=:db_update_placeholder_24, "p7"=:db_update_placeholder_25, "p8"=:db_update_placeholder_26, "p9"=:db_update_placeholder_27, "route_param_key"=:db_update_placeholder_28 WHERE "mlid" = :db_condition_placeholder_0; Array ( [:db_update_placeholder_0] => header-megamenu [:db_update_placeholder_1] => https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff [:db_update_placeholder_2] => [:db_update_placeholder_3] => a:0:@ [:db_update_placeholder_4] => a:1:@s:8:"external";b:1; [:db_update_placeholder_5] => s:13:"VA Leadership"; [:db_update_placeholder_6] => [:db_update_placeholder_7] => 3 [:db_update_placeholder_8] => 1 [:db_update_placeholder_9] => 0 [:db_update_placeholder_10] => menu_link_content:2ff60fae-2cb7-4bb9-a2e5-c7fcf1c4bb59 [:db_update_placeholder_11] => menu_link_content:8a05f6c3-050b-4fa9-8db8-3dd33cccb468 [:db_update_placeholder_12] => Drupal\menu_link_content\Plugin\Menu\MenuLinkContent [:db_update_placeholder_13] => menu_link_content [:db_update_placeholder_14] => a:1:@s:9:"entity_id";s:3:"934"; [:db_update_placeholder_15] => \Drupal\menu_link_content\Form\MenuLinkContentForm [:db_update_placeholder_16] => 0 [:db_update_placeholder_17] => 0 [:db_update_placeholder_18] => 3 [:db_update_placeholder_19] => 1404 [:db_update_placeholder_20] => 1489 [:db_update_placeholder_21] => 1601 [:db_update_placeholder_22] => 0 [:db_update_placeholder_23] => 0 [:db_update_placeholder_24] => 0 [:db_update_placeholder_25] => 0 [:db_update_placeholder_26] => 0 [:db_update_placeholder_27] => 0 [:db_update_placeholder_28] => [:db_condition_placeholder_0] => 1601 ) in Drupal\Core\Entity\Sql\SqlContentEntityStorage->save() (line 815 of /var/www/cms/docroot/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorage.php). [dd.trace_id=6067321073126737708 dd.span_id=1918468261515698340]
```
TLDR from @ndouglas  The actual error was that the URL was too long for the database field in which it was expected to be stored.

@dsasser Did some great research on this and discovered the likely problem. 

> There is a "hidden" 255 character limit here that is causing the error. The link uri field limit is 2048, whereas the menu_tree url field limit is 255. There is a core patch to increase the 255 limit to 2048. https://www.drupal.org/project/drupal/issues/3106205

## AC / Expected behavior
When I click save, even with a really long url, it saves.

## Additional context
Add any other context about the problem here. Reach out to the Product Managers to determine if it should be escalated as critical (**prevents users from accomplishing their work with no known workaround and needs to be addressed within 2 business days**).

5/23 edit
Swirt's thought's: an editor is blocked from doing her job with no known workaround.  In hindsight this should have been a critical defect

### PR DESCRIPTION
**Pull request by edmund-dunn**
## Description

Closes #13278.

## Testing done
Tested locally, link saved correctly without error. 

## QA steps

What needs to be checked to prove this works?  
What needs to be checked to prove it didn't break any related things?  
What variations of circumstances (users, actions, values) need to be checked?  

1. Go to `/admin/structure/menu/item/934/edit?destination=/admin/structure/menu/manage/header-megamenu`
 and paste the following url in the link field `https://department.va.gov/biographies/?_bio_position=assistant-secretary%2Cchairman%2Cdeputy-secretary%2Cgeneral-counsel%2Cinspector-general%2Cprincipal-executive-director%2Csecretary%2Cunder-secretary%2Cva-chief-of-staff%2F%3Fredirect%3D1%2Cva-chief-of-staff`
then click save
   - [x] Validate that the link saves correctly

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by ndouglas — APPROVED** [review]
QA steps worked, verified that no errors occurred, saw nothing new or concerning in the log, link was updated and sent me to the correct URL. Approved pending tests 🙂
